src/main/java/com/spotify/docker/BuildMojo.java

```diff
@@ -493,6 +539,38 @@ private void createDockerFile(final String directory, final List<String> filesTo
     if (maintainer != null) {
       commands.add("MAINTAINER " + maintainer);
     }
+
+    if (env != null) {
+      final List<String> sortedKeys = Ordering.natural().sortedCopy(env.keySet());
+      for (String key : sortedKeys) {
+        final String value = env.get(key);
+        commands.add(String.format("ENV %s %s", key, value));
+      }
+    }
+
+    if (workdir != null) {
+      commands.add("WORKDIR " + workdir);
+    }
+
+    for (String file : filesToAdd) {
+      commands.add(String.format("ADD %s %s", file, file));
+    }
+
+    if (runList != null && !runList.isEmpty()) {
+      for (final String run : runList) {
+        commands.add("RUN " + run);
```

Comment:
@dawidmalina True, but if that happens it's because the end-user has specified a bunch of RUN commands. They can choose to reduce the number of commands or squash them however they'd like.